text/0000-app-image-extensions.md

```diff
@@ -0,0 +1,182 @@
+# Meta
+[meta]: #meta
+- Name: App Image Extensions
+- Start Date: 2019-08-09
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes:
+1. An interface between a set of stack images and a platform in order to extend the images
+2. A UX for the pack CLI for extending builders with OS packages
+
+# Motivation
+[motivation]: #motivation
+
+Allowing buildpacks to install OS packages dynamically during build would drastically increase build time, especially when CNB tooling is used to build or rebase many apps with similar package requirements.
+
+Mixins already allow buildpack authors to create buildpacks that depend on an extended set of OS packages without affecting build time.
+However, it is not uncommon for application code to depend on OS packages.
+
+Given that app authors and platform maintainers experience increased build time directly, extending the stack at the app author's or platform maintainer's request may add flexibility without sacrificing performance.
+
+# What it is
+[what-it-is]: #what-it-is
+
+## Specification
+
+Stack image creators may add the following executables to their run and/or build images:
+
+```
+/cnb/image/build/extend (build-metadata-toml-file) (pkg-cache-dir) | cwd: /
+/cnb/image/run/extend (run-metadata-toml-file) (pkg-cache-dir) | cwd: /
+```
+* executes on base run or build images
+* writes metadata to `/cnb/image/build/metadata.toml` or `/cnb/image/run/metadata.toml`
+* on subsequent builds, lookup latest upstream base image digest:
```

Comment:
What is the `upstream base image here`?

Reply:
The latest available version of the unextended run image.